### DISCUSSION TITLE
Navigation buttons

### DISCUSSION THREAD

**Discussion by somero-77** [sumatrapdfreader/sumatrapdf#4027] [Ideas]
Just a wish of the feature. Very useful feature (as for me), maybe not easy to implement.
Navigation buttons!
Not [leaf the page] [back] and [fore] buttons,
but exactly to the previous/next viewed place in PDF-document.
Sample use scenary: We make a jump to some place in document using the Content/Bookmarks pane, then read through (scroll fore) the text, then we decide to use thet pane again and jump to another place in document, and finaly we want go back to previous read place.
Also useful to combine this feature with quick creation/deletion users bookmarks, with including them in this document browsing history (and maybe also in that navigation pane)

**Comment by kjk** [comment]
We already have that. 

When you click on a inter-page link or in Bookmarks tree, sumatra saves that position and creates a navigation history.

You can navigate those with `Alt + Left` and `Alt + Right`. 

Those are `CmdNavigateBack` and `CmdNavigateForward` commands and you can bind them to a different keyboard shortcut https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts

**Comment by somero-77** [comment]
Thank You for the answer!
 
I checked, what you said, and.. this works.. in a strange way —
firstly, only if «focus» at the document page (a page is clicked) — I think Action Keys have to work independent,
and.. secondly, it seems some segues is remembered, others not.
Besides of that (IMHO):
- there must be screen buttons for that actions — «Go Back» and «Go Fore» — at the tools panel, where the «Previous page» and «Next page» buttons are, as well as "Previous found" and "Next found";
- also the places in document, wich was added to Favorites, MUST have to be added in navigations sequence;
- visibility of Favorites panel itself has to be added in menu.
For example in a such manner:
 
Sincelery, Ilya.
  
>Sun, Feb 4, 2024, 11:47 +02:00 от Krzysztof Kowalczyk ***@***.***>:
>We already have that. When you click on a inter-page link or in Bookmarks tree, sumatra saves that position. You can navigate those with Alt + Left and Alt + Right. Those are CmdNavigateBack and CmdNavigateForward commands and you can bind them to a different keyboard shortcut  https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts

**Comment by GitHubRulesOK** [comment]
- Historic goback/forwardto can only function within context so ONLY apply when document is currently in focus 
and logically forward step only works after you go back as it cannot be predictive.

- *"there must be screen buttons"* that is a continual request which if EVERYBODY had their request there would be no room for any documents (I cannot remove the dozens I dont want or need in acrobat and now edge adds more and more against my desires).
- The current icon set has had few changes over 17 years for the very reason it worked and still does as a lightweight viewer.
- for favorites there are many menu options
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/b17f8e5e-cc44-498f-b1df-4b246bcf5ac9)![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/73c4a9c0-3f45-4b77-a7e1-42d603af1781)![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/e1223dad-d489-4e11-b478-1c56d6df1fa9)![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/9a170e1e-c456-4953-b18b-3ade696faba7)
